### DISCUSSION TITLE
Week6 各项目详情

### DISCUSSION THREAD

**Discussion by zhangchi0923** [creatorsand/.github#24] [Ideas]
能否请实践项目的各位老师介绍下各项目内容、用到的技术、最终的产出等细节，方便学员提前了解和选择。

**Comment by zlemon819** [comment]
+1

**Comment by shiyivei** [comment]
web项目开发：以下面这个Rust AI 图片索引网站应用为例。大家可以选一个实际场景，做一个简单一点web应用，功能上做一些基本的增删改查即可，大体逻辑可以参考下面这样去设计

项目名称：Rust AI 图片索引网站

项目描述：开发一个AI图片索引网站，用户可以上传图片，然后AI会自动分析图片的内容并对其进行标签化，用户可以根据标签搜索相关图片。

技术栈：

1.语言：Rust
2. Web框架：Rocket，这是一个强大的Rust web框架，其特点是类型安全、简单易用。
3. ORM库：Diesel，这是Rust的一个ORM库，可以方便的处理数据库的操作。
4. 模板引擎：Tera，一个由Rust编写的模板引擎，用于生成HTML页面。
5. 数据库：PostgreSQL，一个强大的开源关系型数据库。
6. AI框架：Tract，这是一个Rust写的推理框架，可以用于图像识别。

项目架构：

- 前端：通过Tera模板引擎来生成HTML页面，页面通过CSS和JavaScript进行美化和交互。
- 后端：后端使用Rocket web框架，根据业务逻辑定义路由和处理请求。
- 数据库：后端通过Diesel库来处理与PostgreSQL数据库的交互。
- AI分析：使用Tract进行图片内容的AI分析。

开发步骤：

1. 需求分析：定义图片索引网站的基本功能，比如图片上传，AI分析，图片搜索等。
2. 数据库设计：设计数据库表格和关系，包括图片表，标签表，图片-标签关联表等。
3. AI模型训练：使用Tract和已有的训练数据来训练图片识别的AI模型。
4. 后端开发：使用Rocket和Diesel进行后端开发，实现API接口。
5. 前端开发：使用Tera模板引擎进行前端开发，生成HTML页面，实现与用户的交互。
6. 集成测试：将前端和后端进行集成测试，确保系统功能正确无误。
7. 部署上线：将系统部署到服务器上，进行公网测试。

**Reply by xiegudong45** [reply]
有兴趣加入

**Reply by jiantao88** [reply]
有兴趣

**Reply by BkunS** [reply]
有兴趣

**Comment by bonnie0519** [comment]
web 应用 -> 十一
solana rust 合约 -> Davi
sui Rust 合约 -> uvd
小型区块链项目 -> demian
cosmwasm -> frank

**Comment by Demian101** [comment]
### PLONK 协议优化

参考代码：https://github.com/adria0/plonk-by-fingers
优化目标：目前的 Copy constrains 是写死的，可以尝试对其进行优化

**Comment by vladilen11** [comment]
## CosmWasm 合约应用
**官方链接：https://cosmwasm.com/**

CosmWasm 是为Cosmos 生态系统建造的一个新型智能合约平台。如果你从未听说过它，请先阅读这篇入门级文章：[CosmWasm宣布正式启动](https://blog.cosmos.network/announcing-the-launch-of-cosmwasm-cc426ab88e12)。
CosmWasm 是Rust语言合约，开发者需要用Rust开发Cosmwasm 合约，并将其编译成wasm文件部署到链上。

## 学习资料
- <https://book.cosmwasm.com/>
- <https://forum.cosmos.network/t/cosmwasm/3083>
- <https://academy.cosmwasm.com/>

**Comment by DaviRain-Su** [comment]
# Solana 合约开发方向参考

- 简单的swap， fork一下官方代码，做一些定制化
- 简单的orderbook, fork一下serum ，做一些定制化
- gamefi 猜大小的等游戏 
- nft 市场，批量扫的工具等- 
-  一键发nft/token的工具  
-  帮助发空投的工具

**Reply by DaviRain-Su** [reply]
- Anchor docs: https://www.anchor-lang.com/docs/installation
- Solana docs: https://docs.solana.com/
-
